internal/controller/apiextensions/composite/composition_pt.go

```diff
@@ -279,7 +279,7 @@ func (c *PTComposer) Compose(ctx context.Context, xr *composite.Unstructured, re
 				// run again the composition after some other resource is
 				// created or updated successfully. So, we emit a warning event
 				// and move on.
-				events = append(events, event.Warning(reasonCompose, errors.Wrap(err, errApplyComposed)))
+				events = append(events, event.Warning(reasonCompose, errors.Wrapf(err, errApplyComposed, ptr.Deref(t.Name, fmt.Sprintf("%d", i+1)))))
```

Comment:
I _think_ we might fall back to the array index in some other messages where `t.Name` is `nil`, but I'm not sure we use `i+1` - we might just use `I`. Can you double check to make sure this will be consistent?

Reply:
Thanks for checking!